metals/src/main/scala/scala/meta/internal/metals/Doctor.scala

```diff
@@ -72,7 +72,7 @@ final class Doctor(
       clientCommand: Command,
       onServer: MetalsHttpServer => Unit
   ): Unit = {
-    if (clientConfig.isExecuteClientCommandProvider) {
+    if (clientConfig.isExecuteClientCommandProvider && !clientConfig.isHttpEnabled) {
```

Comment:
In this scenario I think that `isHttpEnabled` should win. Mainly due to the fact that while all clients _should_ support all the commands that the server can send, they may not. For example in nvim-metals, we are making the client as `isExecuteClientCommandProvider` to true because we need some stuff (for the new scala file command), but we still want the doctor to be in the browser. Without this change, doing that is impossible.